### PR TITLE
add 'git clone' step to quickstart guides

### PR DESCRIPTION
**Pull request by alexcpn**
Signed-off-by: AlexPunnen <alex.punnen@nokia.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[skip ci]

### CODE REVIEW

**Comment by alexcpn** [comment]
@BlaineEXE @travisn  PR from https://github.com/rook/rook/pull/3497

**Comment by alexcpn** [comment]
Hi, I could do that. I was trying to play with the markup templating Jekyll and Liquid used; However, these are not currently rendered in Github for rook documentation ex [ceph-quickstart] page has all these variables unparsed and as is (https://github.com/rook/rook/blob/master/Documentation/ceph-quickstart.md) 
(I am assuming everyone is checking these files for documentation and not something generated out of this. At least I was referring directly to these markup links)

IMO adding these markups that do not render, makes the documentation messier without proper substitution not happening. I can add a comment describing the same in words if that's fine.

**Comment by alexcpn** [comment]
I will add . I  will check if there is a way to mask the jekyll directives that are rendered by githubpages for  https://rook.io/docs/rook/v1.0/ceph-quickstart.html, to not interfere here - https://github.com/rook/rook/blob/master/Documentation/ceph-quickstart.md

If not I will just do it like current

**Comment by travisn** [comment]
@alexcpn Yeah, would be great if we could find how to hide that in the github rendering, but users should all be looking at the rook.io docs anyway so it has been fine.

**Comment by alexcpn** [comment]
I could not find an easy way to do this , also did not have the time to check more; But to reduce the ugliness a bit and avoid repetition added the script in a liquid file and added a relative include. (I am new in Jekyll and the convention is to have include file in _include directory which is not used in this project. 
This relative include {% include_relative branch.liquid %} seems to work in github
Example
https://github.com/alexcpn/alexcpn.github.io/tree/master/_posts
https://alexcpn.github.io/2019/08/21/Second-Post.html

File: Documentation/branch.liquid [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% assign url = page.url | split: '/' %}
```
Comment (travisn) [inline Documentation/branch.liquid]: Ah, nice to have the include file so we can stop replicating. Note that the rook docs can be generated by the instructions [here](https://github.com/rook/rook.github.io#test). Could you verify with those steps? If not, I'll try to get back to it in the next couple days to verify. Thanks!

File: Documentation/cassandra.md [hunk 2]
```diff
@@ -20,6 +21,7 @@ To make sure you have a Kubernetes cluster that is ready for `Rook`, you can [fo
 First deploy the Rook Cassandra Operator using the following commands:
 
 ```console
+git clone  --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (travisn) [inline Documentation/cassandra.md]: nit: these `git clone` commands have double spaces between some of the args

**Comment by galexrt** [comment]
Ping @alexcpn

**Comment by alexcpn** [comment]
> Please add a commit message. And next time, use the `[skip ci]` flag to not run the CI. Thanks.

Commit messages are added

File: Documentation/branch.liquid (hunk 1 above)
Comment (alexcpn) [reply Documentation/branch.liquid]: there is no target name 'run' in the main MakeFile or the one in build/release.

**Comment by travisn** [comment]
@alexcpn A few final requests, sorry for the delay:
- There are some merge conflicts. Could you rebase on the latest?
- Squash the commits and add a more descriptive commit message. The format required by our new bot would be like this:
```
docs: add 'git clone' step to quickstart guides

Add documentation to clone the correct release branch of the repo
to reduce confusion from pulling the wrong branch.
```

File: Documentation/branch.liquid (hunk 1 above)
Comment (alexcpn) [reply Documentation/branch.liquid]: > @alexcpn A few final requests, sorry for the delay:
> 
> * There are some merge conflicts. Could you rebase on the latest?
> * Squash the commits and add a more descriptive commit message. The format required by our new bot would be like this:
> 
> ```
> docs: add 'git clone' step to quickstart guides
> 
> Add documentation to clone the correct release branch of the repo
> to reduce confusion from pulling the wrong branch.
> ```

done

**Comment by alexcpn** [comment]
> Please add a commit message. And next time, use the `[skip ci]` flag to not run the CI. Thanks.

done

File: Documentation/branch.liquid (hunk 1 above)
Comment (alexcpn) [reply Documentation/branch.liquid]: I could not test this locally; I have tested this Liquid templating in my github acc some time back however

**Comment by alexcpn** [comment]
@travisn Could you do a re-review; or anything you want me to do regarding this

File: Documentation/ceph-quickstart.md [hunk 3]
```diff
@@ -28,9 +24,11 @@ If you are using `dataDirHostPath` to persist rook data on kubernetes hosts, mak
 
 ## TL;DR
 
-If you're feeling lucky, a simple Rook cluster can be created with the following kubectl commands and [example yaml files](https://github.com/rook/rook/blob/{{ branchName }}/cluster/examples/kubernetes/ceph). For the more detailed install, skip to the next section to [deploy the Rook operator](#deploy-the-rook-operator).
+If you're feeling lucky, a simple Rook cluster can be created with the following kubectl commands and [example yaml files] 
+https://github.com/rook/rook/blob/ {{ branchName }} /cluster/examples/kubernetes/ceph. For the more detailed install, skip to the next section to [deploy the Rook operator](#deploy-the-rook-operator).
```
[the remark this answers is not in the document]
Comment (alexcpn) [reply Documentation/ceph-quickstart.md]: corrected

**Comment by alexcpn** [comment]
@travisn  - all comments incorporated

File: Documentation/cassandra.md [hunk 4]
```diff
@@ -20,6 +21,7 @@ To make sure you have a Kubernetes cluster that is ready for `Rook`, you can [fo
 First deploy the Rook Cassandra Operator using the following commands:
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/cassandra.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/ceph-monitoring.md [hunk 5]
```diff
@@ -37,6 +38,7 @@ With the Prometheus operator running, we can create a service monitor that will
 From the root of your locally cloned Rook repo, go the monitoring directory:
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/ceph-monitoring.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/ceph-quickstart.md [hunk 6]
```diff
@@ -3,13 +3,9 @@ title: Ceph Storage
...
 indent: true
 ---
-{% assign url = page.url | split: '/' %}
-{% assign currentVersion = url[3] %}
-{% if currentVersion != 'master' %}
-{% assign branchName = currentVersion | replace: 'v', '' | prepend: 'release-' %}
-{% else %}
-{% assign branchName = currentVersion %}
-{% endif %}
+
```
Comment (leseb) [inline Documentation/ceph-quickstart.md]: Unnecessary space?

File: Documentation/ceph-quickstart.md [hunk 7]
```diff
@@ -31,6 +27,7 @@ If you are using `dataDirHostPath` to persist rook data on kubernetes hosts, mak
 If you're feeling lucky, a simple Rook cluster can be created with the following kubectl commands and [example yaml files](https://github.com/rook/rook/blob/{{ branchName }}/cluster/examples/kubernetes/ceph). For the more detailed install, skip to the next section to [deploy the Rook operator](#deploy-the-rook-operator).
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/ceph-quickstart.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/cockroachdb.md [hunk 8]
```diff
@@ -19,6 +20,7 @@ To make sure you have a Kubernetes cluster that is ready for `Rook`, you can [fo
 First deploy the Rook CockroachDB operator using the following commands:
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/cockroachdb.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/edgefs-csi.md [hunk 9]
```diff
@@ -105,6 +106,7 @@ By using `k8sEdgefsNamespaces` and `k8sEdgefsMgmtPrefix` parameters, driver is c
 Check configuration options and create kubernetes secret for Edgefs CSI NFS plugin
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/edgefs-csi.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/edgefs-monitoring.md [hunk 10]
```diff
@@ -4,6 +4,8 @@ weight: 4800
 indent: true
 ---
```
Comment (leseb) [inline Documentation/edgefs-monitoring.md]: Unnecessary space.

File: Documentation/edgefs-monitoring.md [hunk 11]
```diff
@@ -34,6 +36,7 @@ With the Prometheus operator running, we can create a service monitor that will
 From the root of your locally cloned Rook repo, go the monitoring directory:
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/edgefs-monitoring.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/edgefs-quickstart.md [hunk 12]
```diff
@@ -4,6 +4,8 @@ weight: 550
 indent: true
 ---
```
Comment (leseb) [inline Documentation/edgefs-quickstart.md]: Unnecessary space.

File: Documentation/edgefs-quickstart.md [hunk 13]
```diff
@@ -42,6 +44,7 @@ To turn off this node adjustment need to enable `skipHostPrepare` option in clus
 If you're feeling lucky, a simple EdgeFS Rook cluster can be created with the following kubectl commands. For the more detailed install, skip to the next section to [deploy the Rook operator](#deploy-the-rook-operator).
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/edgefs-quickstart.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/minio-object-store.md [hunk 14]
```diff
@@ -4,6 +4,8 @@ weight: 600
 indent: true
 ---
```
Comment (leseb) [inline Documentation/minio-object-store.md]: Unnecessary space.

File: Documentation/minio-object-store.md [hunk 15]
```diff
@@ -20,6 +22,7 @@ To make sure you have a Kubernetes cluster that is ready for `Rook`, you can [fo
 First deploy the Rook Minio operator using the following commands:
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/minio-object-store.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/nfs.md [hunk 16]
```diff
@@ -22,6 +23,7 @@ You can read further about the details and limitations of these volumes in the [
 First deploy the Rook NFS operator using the following commands:
 
 ```console
+git clone --single-branch --branch  {{ branchName }} https://github.com/rook/rook.git
```
Comment (leseb) [inline Documentation/nfs.md]:
```suggestion
git clone --single-branch --branch {{ branchName }} https://github.com/rook/rook.git
```

File: Documentation/cassandra.md (hunk 4 above)
Comment (alexcpn) [reply Documentation/cassandra.md]: all done
